src/date-utils.es6

```diff
@@ -156,21 +156,86 @@ const DateUtils = {
     return moment(date)
   },
 
+
+  /**
+   * Return a formatting string for displaying time
+   *
+   * @param {Date} opts - Object with different properties for customising output
+   * @return {String} The format string based on syntax used by Moment.js
+   *
+   * seconds, upperCase and timeZone are the supported extra options to the format string.
+   * Checks whether or not to use 24 hour time format.
+   */
   getTimeFormat(opts) {
     const use24HourClock = NylasEnv.config.get('core.workspace.use24HourClock')
     let timeFormat = use24HourClock ? "HH:mm" : "h:mm"
+
     if (opts && opts.seconds) {
       timeFormat += ":ss"
     }
-    if (!use24HourClock && opts && opts.upperCase) {
-      timeFormat += " A"
+
+    // Append meridian if not using 24 hour clock
+    if (!use24HourClock) {
+      if (opts && opts.upperCase) {
+        timeFormat += " A"
+      } else {
+        timeFormat += " a"
+      }
     }
+
     if (opts && opts.timeZone) {
       timeFormat += " z"
     }
 
     return timeFormat
   },
+
+
+  /**
+   * Return a short format date/time
+   *
+   * @param {Date} datetime - Timestamp
+   * @return {String} Formated date/time
+   *
+   * The returned date/time format depends on how long ago the timestamp is.
+   */
+  shortTimeString(datetime) {
+    const diff = moment().diff(datetime, 'days', true)
+    let format = null
+
+    if (diff <= 1) {
```

Comment:
I think we actually want to pull the more advanced logic from the function above into here:

```
 -      isSameDay = now.isSame(msgDate, 'days')		
 -      if diff < 1 and isSameDay		
 -        return msgDate.format timeFormat		
 -      if diff < 1.5 and not isSameDay		
 -        timeAgo = msgDate.from now		
 -        monthAndDay = msgDate.format timeFormat		
 -        return monthAndDay + " (" + timeAgo + ")"		
```

Basically, this changes the behavior a bit:

- Email sent 0-24 hours ago: Just the time (eg: 9AM)
- Email sent 24-36 hours ago: Both the time and the date "July 17 (1AM)"
- Email sent 36+ hours ago: Just the date

I think having that intermediate state is important because people still want the time resolution. I believe it was added in a PR a while back!